src/components/editor/EditAuthorList.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import _ from 'lodash';
+import { shallowEquals } from 'lib/BaseComponent';
+
+export default class EditAuthorList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      // Key is slug, value is name
+      authorsAdded: {},
+      // An object where key is slug and value is whether it's deleted or not
+      authorsDeleted: this.props.data.reduce((cur, next) => {
+        cur[next.slug] = false;
+        return cur;
+      }, {})
+    };
+    this.handleAddAuthor = this.handleAddAuthor.bind(this);
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return (!shallowEquals(this.state.authorsAdded, nextState.authorsAdded) || 
+      !shallowEquals(this.state.authorsDeleted, nextState.authorsDeleted) ||
+      !shallowEquals(this.props.data, nextProps.data));
+  }
+
+  componentDidUpdate() {
+    this.props.onChange(this.state.authorsAdded, this.state.authorsDeleted);
+  }
+
+  handleAddAuthor() {
+    let newValue = {};
+    newValue[this.addAuthorNode.value] = this.addAuthorNode.value;
+    let newAuthorsAdded = Object.assign({}, this.state.authorsAdded, newValue);
+    this.setState({
+      authorsAdded: newAuthorsAdded
+    });
+    // Clear input field
+    this.addAuthorNode.value = "";
+  }
+
+  handleDeleteAuthor(slug, originalAuthor) {
+    if (originalAuthor) {
+      let newValue = {};
+      newValue[slug] = true;
+      let newAuthorsDeleted = Object.assign({}, this.state.authorsDeleted, newValue);
+      this.setState({
+        authorsDeleted: newAuthorsDeleted
+      });
+    }
+    else {
+      let copy = Object.assign({}, this.state.authorsAdded);
+      delete copy[slug];
```

Comment:
No because delete actually changes the original object in-place in-memory. So when it compares, its comparing identical objects. So it won't re-render.